_posts/2024-02-01-wildfly-glow.adoc

```diff
@@ -0,0 +1,509 @@
+
+=== WildFly Provisioning diagram
+
+image::wildfly-glow1.jpg[WildFly provisioning diagram]
+
+The above shows a simplified representation of what provisioning is. Let's look at the complexity that comes with such flexibility.
+
+== Current WildFly Provisioning has a complex configuration
+
+The issues we are currently facing are mainly located at the user configuration level. 
+It is not that trivial to build-up a proper WildFly server to run your applications.
+
+Known issues:
+
+* How to know about WildFly compatible Galleon feature-packs?
+* How to discover the Galleon layers that my application requires to properly work?
+
+Current solutions:
+
+* Use https://docs.wildfly.org/30/[WildFly documentation], look at https://github.com/wildfly/quickstart[WildFly quickstarts], 
+search for blogs and/or github projects to discover extra feature-packs.
+
+* Use WildFly base layers (aggregators for the main application use cases) such as `cloud-server` or `jaxrs-server` layers. 
+Something to note is that when using these aggregators the set of features that you are provisioning is not optimal. 
+More content is generally provisioned than what your application requires. Further-more these layers don't contain it all, according to your application requirements,
+you would need to add some more layers (e.g.: `microprofile-config`, `ejb`, `jsf`, ...).
+
+To summarize we need a way to discover all the extra Galleon feature-packs that would provide interesting features and a way to link 
+the deployed application(s) to the set of required Galleon layers.
+
+This is what https://github.com/wildfly/wildfly-glow[WildFly Glow] is all about, a provisioning time bridge between the deployment and the WildFly server.
+
+== WildFly Glow
+
+Glow stands for “Galleon Layers Output from War”. By inspecting the deployment(s), WildFly Glow 
+can produce the set of Galleon feature-packs and layers that your application requires.
+
+WildFly Glow knows the set of extra feature-packs compatible with a given WildFly version. 
+It understands the connection that exists between Galleon layers and your application:
+
+* Java types and annotations in use
+* XML descriptors,
+* Properties files, …
+
+WildFly Glow can suggest interesting features not directly required by your application but meaningful. For example: SSL when undertow is in use, 
+Microprofile OpenAPI when jaxrs is in use, WildFly CLI(s) (all CLI launch scripts located in the WILDFLY_HOME/bin directory), ...
+
+WildFly Glow goes one step further than pure provisioning by identifying potential errors. When errors are identified, 
+it can suggest ways to fix them (eg: add a missing datasource, add a missing access to JMS broker).
+
+To summarize, WildFly Glow can build the server from your deployments and helps you discover additional 
+WildFly server features that you are perhaps not aware of.
+
+WildFly Glow detailed http://docs.wildfly.org/wildfly-glow[documentation]. 
```

Comment:
Perhaps

"The WildFly documentation site includes http://docs.wildfly.org/wildfly-glow[detailed documentation] of WildFly Glow.